client/src/components/geneExpression/geneSet.js

```diff
@@ -0,0 +1,187 @@
+    const { dispatch, setName } = this.props;
+    dispatch({
+      type: "color by gene set",
+      colorAccessor: setName,
+    });
+  };
+
+  toggleSummaryHisto = () => {
+    const { toggleSummaryHisto } = this.state;
+    this.setState({ toggleSummaryHisto: !toggleSummaryHisto });
+  };
+
+  render() {
+    const { setName, setGenes, isDiffexp } = this.props;
+    const { isOpen, toggleSummaryHisto } = this.state;
+    return (
+      <div style={{ marginBottom: 3 }}>
+        <div
+          style={{
+            display: "flex",
+            justifyContent: "space-between",
+            alignItems: "baseline",
+          }}
+        >
+          <span
+            role="menuitem"
+            tabIndex="0"
+            data-testclass="geneset-expand"
+            data-testid={`${setName}:geneset-expand`}
+            onKeyPress={/* todo_genesets */ () => {}}
+            style={{
+              cursor: "pointer",
+              userSelect: "none",
+            }}
+            onClick={this.onGenesetMenuClick}
+          >
+            <Truncate>
+              <span
+                style={{
+                  maxWidth:
+                    globals.leftSidebarWidth -
+                    120 /* todo_genesets this magic number determines how much of a long geneset name we see, and will be tweaked as we build */,
+                }}
+                data-testid={`${setName}:geneset-label`}
+              >
+                {setName}
+              </span>
+            </Truncate>
+            {isOpen ? (
+              <FaChevronDown
+                data-testclass="geneset-expand-is-expanded"
+                style={{ fontSize: 10, marginRight: 5 }}
+              />
+            ) : (
+              <FaChevronRight
+                data-testclass="geneset-expand-is-not-expanded"
+                style={{ fontSize: 10, marginRight: 5 }}
+              />
+            )}
+          </span>
+          <div>
+            <GenesetMenus genesetsEditable geneset={setName} />
+          </div>
+        </div>
+
+        <div style={{ marginLeft: 15, marginTop: 5, marginRight: 0 }}>
+          {isOpen ? (
+            <Tooltip
+              content="Aggregate all genes in this geneset, and allow coloring by and selecting on the entire set."
+              position={Position.BOTTOM_RIGHT}
+              usePortal
+              hoverOpenDelay={globals.tooltipHoverOpenDelay}
+              modifiers={{
+                preventOverflow: { enabled: false },
+                hide: { enabled: false },
+              }}
+            >
+              <Switch
+                style={{ fontStyle: "italic" }}
+                checked={toggleSummaryHisto}
+                label="Show mean expression across entire set"
+                alignIndicator="left"
+                innerLabel="off"
+                innerLabelChecked="on"
+                onChange={this.toggleSummaryHisto}
+              />
+            </Tooltip>
+          ) : null}
```

Comment:
```suggestion
          {isOpen && (
            <Tooltip
              content="Aggregate all genes in this geneset, and allow coloring by and selecting on the entire set."
              position={Position.BOTTOM_RIGHT}
              usePortal
              hoverOpenDelay={globals.tooltipHoverOpenDelay}
              modifiers={{
                preventOverflow: { enabled: false },
                hide: { enabled: false },
              }}
            >
              <Switch
                style={{ fontStyle: "italic" }}
                checked={toggleSummaryHisto}
                label="Show mean expression across entire set"
                alignIndicator="left"
                innerLabel="off"
                innerLabelChecked="on"
                onChange={this.toggleSummaryHisto}
              />
            </Tooltip>
          )}
```

Reply:
@seve I disagree that this is more clear. @maniarathi, tiebreaker?

Reply:
I originally did it the same way, but Timmy pointed out this version to me. With a ternary its expected that there's a second half (if/else).  While the `&&` more clearly denotes that it's `if...then`

Reply:
I agree with @seve.   Ternary is confusing in this case.

Reply:
Ok, search shows `src/client/components` use this pattern upwards of 60 times. I'll fix here and create a ticket for tech debt.